runtime/lua/vim/lsp/util.lua

```diff
@@ -149,16 +149,42 @@ function M.apply_text_edits(text_edits, bufnr)
     table.insert(lines, '')
   end
 
+  -- store current cursor pos
+  local curpos_fixed = false
+  local curpos = api.nvim_win_get_cursor(0)
```

Comment:
What happens if the bufnr is not for the current window -- I think we may want to additionally save what window we're currently in, so we can go back to it.

Reply:
Oh... maybe it doesn't matter because you check the bufnr at the end. I'm not sure.

Reply:
You're correct. I think it might be good to avoid calling on the no current buffer but It makes ugly code, I think.

I would like to add the document here for now.